sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala

```diff
@@ -842,10 +843,12 @@ private[spark] class HiveExternalCatalog(conf: SparkConf, hadoopConf: Configurat
     // source tables. Here we set the table location to `locationUri` field and filter out the
     // path option in storage properties, to avoid exposing this concept externally.
     val storageWithLocation = {
-      val tableLocation = getLocationFromStorageProps(table)
+      val tableLocation = getLocationFromStorageProps(table).map { loc =>
+        val uri = CatalogUtils.stringToURI(loc)
+        table.storage.locationUri.map(HiveExternalCatalog.toAbsoluteURI(uri, _)).getOrElse(uri)
```

Comment:
`table.storage.locationUri` is the table location, not database location, isn't it?

Reply:
The table location uri has been rewritten by HiveClientImpl.convertHiveTableToCatalogTable.
For example:
```
table.storage.locationUri: Some(viewfs://clusterA/user/hive/warehouse/db_spark_39203.db/t1)
loc: /user/hive/warehouse/db_spark_39203.db/t1
```

Reply:
Sorry I don't understand. The previous code completely ignores `table.storage.locationUri`. Do you mean we still don't care about `table.storage.locationUri`, but only want to get its qualifier (scheme, host, port, etc.)?